R/cache_postgresql.R

```diff
@@ -0,0 +1,138 @@
+#' PostgreSQL Cache
+#' PostgreSQL-backed cache, for remote caching.
+#'
+#' Create a table with key and val columns
+#' CREATE TABLE r_cache (
+#'   key varchar(128),
+#'   val text
+#' )
+#'
+#' @examples
+#'
+#' \dontrun{
+#' pg <- cache_postgresql(pg_con, "r_cache")
+#' mem_runif <- memoise(runif, cache = pg)
+#' }
+#'
+#'
+#' @param pg_con An RPostgreSQL connection object
+#' @param table_name A postgres table name with val and key fields. May include a schema.
+#' @param compress Argument passed to \code{saveRDS}. One of FALSE, "gzip",
+#' "bzip2" or "xz". Default: FALSE.
+#' @inheritParams cache_memory
+#' @export
+
+cache_postgresql <- function(pg_con, table_name, algo = "sha512", compress = FALSE) {
+
+  path <- tempfile("memoise-")
+  dir.create(path)
+
+  if (!(requireNamespace("RPostgreSQL"))) { stop("Package `RPostgreSQL` must be installed for `cache_postgresql()`.") } # nocov
+  if (!(requireNamespace("DBI"))) { stop("Package `DBI` must be installed for `cache_postgresql()`.") } # nocov
+  if (!(requireNamespace("readr"))) { stop("Package `readr` must be installed for `cache_postgresql()`.") } # nocov
+
+  cache_reset <- function() {
+    dbSendQuery(
```

Comment:
The tests are failing from R packaging, e.g. `DBI::dbSendQuery` needs to be added to all the imported functions used